tests/conftest.py

```diff
@@ -66,45 +96,64 @@ async def sandbox_realm(context_id, websocket):
     return result["realm"]
 
 
-@pytest_asyncio.fixture
-async def context_id(websocket):
-    """Return the context id from the first browsing context."""
-    result = await get_tree(websocket)
-    return result["contexts"][0]["context"]
+@pytest.fixture
+def url_same_origin():
+    """Return a same-origin URL."""
+    return 'about:blank'
 
 
-@pytest_asyncio.fixture
-async def another_context_id(create_context):
-    return await create_context()
+# TODO(sadym): make offline.
+@pytest.fixture(params=[
+    'https://example.com/',  # Another domain: Cross-origin
+    'data:text/html,<h2>child page</h2>',  # Data URL: Cross-origin
+])
+def url_cross_origin(request):
+    """Return a cross-origin URL."""
+    return request.param
 
 
-@pytest_asyncio.fixture
-def create_context(websocket):
-    """Return a function that creates a new browsing context."""
+# TODO(sadym): make offline.
+@pytest.fixture(params=[
+    'about:blank',  # Same-origin
+    'https://example.com/',  # Another domain: Cross-origin
+    'data:text/html,<h2>child page</h2>',  # Data URL: Cross-origin
+])
+def url_all_origins(request):
+    """Return an URL exhaustively, including same-origin and cross-origin."""
+    return request.param
 
-    async def _():
-        result = await execute_command(websocket, {
-            "method": "browsingContext.create",
-            "params": {
-                "type": "tab"
-            }
-        })
-        return result['context']
 
-    return _
+@pytest.fixture
+def html():
+    """Return a factory for HTML data URL with the given content."""
 
+    def html(content=""):
+        return f'data:text/html,{content}'
```

Comment:
url encode?

Reply:
This will break client code. There's some downstream code that is quoting manually. Can we do that in a follow-up? For now, I'd like this to be a no-op transformation where possible.